Extension/src/Utility/Eventing/eventParser.ts

```diff
@@ -0,0 +1,262 @@
+                            continue outer;
+
+                        case Kind.Backslash:
+                            // take the next token, and add it to the regex
+                            rxExpression.push(inner.shift()!);
+                            continue;
+                    }
+
+                    // if we have nothing left, and we haven't closed the regex, then it's an error
+                    if (!inner.length) {
+                        throw new Error(`unterminated regular expression ${inner}`);
+                    }
+                }
+            }
+                break;
+
+            case Kind.OpenParen:
+            case Kind.CloseParen:
+            case Kind.BarBar:
+            case Kind.AmpersandAmpersand:
+                // add that token to the final expression
+                expression.push(token.text!);
+                break;
+
+            case Kind.StringLiteral:
+                // if we're given a string literal and the next token after that is one of {||, &&, ]}
+                // then we can just use the string literal as a comparison
+                // otherwise it should be part of the JavaScript expression (i.e. ['foo'===bar])
+                eatWhitespace(inner);
+                const peek = inner[0];
+                if (peek) {
+                    switch (peek.kind) {
+                        case Kind.BarBar:
+                        case Kind.AmpersandAmpersand:
+                        case Kind.CloseBracket:
+                            expression.push(`$strings.has(${token.text!})`);
+                            continue outer;
+                    }
+                }
+
+            // eslint-disable-next-line no-fallthrough
+            default:
+                // anything else, take all the tokens until we hit a separator/terminator/the end
+                const js = [token];
+                while (inner.length) {
+                    token = inner.shift()!;
+                    switch (token.kind) {
+                        case Kind.BarBar:
+                        case Kind.AmpersandAmpersand:
+                        case Kind.CloseBracket:
+                            // push the token back on the stack, and we're done
+                            inner.unshift(token);
+                            break;
+                        default:
+                            js.push(token);
+                            continue;
+                    }
+                    break;
+                }
+                const jsExpression = js.map(each => each.text).join('');
+
+                // sandbox style, assumes $data is set to Event.data and $strings is the collection of strings for the discriminator or the event itself
+                expression.push(jsExpression.trim());
+                break;
+        }
+    }
+
+    // $data -- either the event data or the event source object if there is no event data.

```

Comment:
The 1st line uses `--` and the 2nd uses `-`. Can you use the same style? I don't know which is better. Up to you.

Reply:
fixing